internal/resource/config/authenticationapi/authentication_api_settings_resource.go

```diff
@@ -133,15 +133,19 @@ func (r *authenticationApiSettingsResource) Configure(_ context.Context, req res
 
 }
 
-func readAuthenticationApiSettingsResponse(ctx context.Context, r *client.AuthnApiSettings, state *authenticationApiSettingsResourceModel, expectedValues *authenticationApiSettingsResourceModel, diags *diag.Diagnostics) {
+func readAuthenticationApiSettingsResponse(ctx context.Context, r *client.AuthnApiSettings, state *authenticationApiSettingsResourceModel, expectedValues *authenticationApiSettingsResourceModel) diag.Diagnostics {
 	//TODO different placeholder?
+	var diags diag.Diagnostics
 	state.Id = types.StringValue("id")
 	state.ApiEnabled = types.BoolValue(*r.ApiEnabled)
 	state.EnableApiDescriptions = types.BoolValue(*r.EnableApiDescriptions)
 	state.RestrictAccessToRedirectlessMode = types.BoolValue(*r.RestrictAccessToRedirectlessMode)
 	state.IncludeRequestContext = types.BoolValue(*r.IncludeRequestContext)
-	resourceLinkObjectValue := resourcelink.ToState(ctx, r.DefaultApplicationRef, diags)
+	var valueFromDiags diag.Diagnostics
```

Comment:
Since we only ever generate one diagnostics object in this method, there's no need to create a separate one or declare it ahead of time. You can just do something like 
`
resourceLinkObjectValue, valueFromDiags := resourcelink.ToState(ctx, r.DefaultApplicationRef)
state.DefaultApplicationRef = resourceLinkObjectValue
return valueFromDiags
`